Jenkinsfile

```diff
@@ -198,11 +198,17 @@ pipeline {
           }
           steps {
             withGithubNotify(context: 'Build-Test - OSX') {
-              deleteDir()
-              unstash 'source'
-              dir("${BASE_DIR}"){
-                sh script: './scripts/jenkins/before_install.sh', label: 'Install dependencies'
-                sh script: './scripts/jenkins/build-test.sh', label: 'Build and test'
+              retry(3) {
+                deleteDir()
+                unstash 'source'
+                dir("${BASE_DIR}"){
+                  sh script: './scripts/jenkins/before_install.sh', label: 'Install dependencies'
+                }
+              }
+              retry(3) {
+                dir("${BASE_DIR}"){
+                  sh script: './scripts/jenkins/build-test.sh', label: 'Build and test'
+                }
               }
             }
```

Comment:
This block is almost the same as the one produced `generateStep`, only differing in the labels for the node. @elastic/observablt-robots wdyt about refactoring this reusing the above step, parametrised by `version` and `nodeLabels`?

Reply:
You certainly could, but IMHO this is quite readable as-is and not much would be gained by a refactor.

Reply:
If it's readable then you have my sword :)